rpc/backend/tracing.go

```diff
@@ -54,11 +56,20 @@ func (b *Backend) TraceTransaction(hash common.Hash, config *evmtypes.TraceConfi
 			continue
 		}
 		for _, msg := range tx.GetMsgs() {
-			ethMsg, ok := msg.(*evmtypes.MsgEthereumTx)
-			if !ok {
+			ethMsg, isEth := msg.(*evmtypes.MsgEthereumTx)
+			_, isBankSend := msg.(*banktypes.MsgSend)
+			if !isEth && !isBankSend {
 				continue
 			}
 
+			if isBankSend {
+				ethMsg, err = utilstx.BankSendTxAsEthereumTx(tx)
+				if err != nil {
+					b.logger.Debug("failed to convert bank send transaction into eth tx", "height", blk.Block.Height, "error", err.Error())
+					continue
+				}
+			}
+
 			predecessors = append(predecessors, ethMsg)
```

Comment:
more of a nit, but a slight performance boost - if it's an `MsgEthereumTx` you don't need the type checking for the `MsgSend`. Whether to first check for the Eth transaction or the bank transfer could be decided depending on which of the two is more commonly present on your chain.

```suggestion
			if ethMsg, isEth := msg.(*evmtypes.MsgEthereumTx); isEth {
                predecessors = append(predecessors, ethMsg)
                continue
            }

			if _, isBankSend := msg.(*banktypes.MsgSend); isBankSend {
				ethMsg, err = utilstx.BankSendTxAsEthereumTx(tx)
				if err != nil {
					b.logger.Debug("failed to convert bank send transaction into eth tx", "height", blk.Block.Height, "error", err.Error())
					continue
				}
                predecessors = append(predecessors, ethMsg)
			}
```